src/systempolicy/systemPolicy.go

```diff
@@ -271,14 +271,14 @@ func populateKnoxSysPolicyFromWPFSDb(namespace, clustername, labels, fromsource
 func WriteSystemPoliciesToFile_Ext(namespace, clustername, labels, fromsource string, includeNetwork bool) {
 	kubearmorK8SPolicies := extractK8SSystemPolicies(namespace, clustername, labels, fromsource, includeNetwork)
 	for _, pol := range kubearmorK8SPolicies {
-		fname := "kubearmor_policies_" + pol.Metadata["clusterName"] + "_" + pol.Metadata["namespace"] + "_" + pol.Metadata["containername"] + "_" + pol.Metadata["name"]
+		fname := "kubearmor_policies_" + pol.Metadata.Annotations["clusterName"] + "_" + pol.Metadata.Annotations["namespace"] + "_" + pol.Metadata.Annotations["containername"] + "_" + pol.Metadata.Annotations["name"]
```

Comment:
here i feel that it (namespaces,name etc) should be part of Metadata spec not annotation spec but i would suggest to check with @wazir-ahmed

Reply:
addressed